### PR TITLE
Improve Android and CD docs

### PR DESCRIPTION
**Pull request by guidezpl**
Based on my experience going setting this up for the gallery

### CODE REVIEW

File: src/docs/deployment/cd.md [hunk 1]
```diff
@@ -7,11 +7,33 @@ Follow continuous delivery best practices with Flutter to make sure your
 application is delivered to your beta testers and validated on a frequent basis
 without resorting to manual workflows.
 
+
+## CI/CD Options
+
+There are a number of continuous integration (CI) and continuous delivery (CD)
+options available to help automate the delivery of your application.
+
+### All-in-one type of options, Flutter functionality built in XXX
```
Comment (guidezpl) [inline src/docs/deployment/cd.md:16]: Marked these headers with XXX for extra attention needed

File: src/docs/deployment/cd.md [hunk 2]
```diff
@@ -7,11 +7,33 @@ Follow continuous delivery best practices with Flutter to make sure your
...
+* [Cirrus][]
+    * Example: Flutter framework repository's [Cirrus script][].
+* [GitHub Actions][]
+    * Example: Flutter Gallery's [Github Actions workflows][]
+    * Example: [Github Action in Flutter Project][]
+* [Travis][]
+* [GitLab][]
+
+This guide shows how to set up fastlane and then integrate it with 
+your existing testing and continuous integration (CI) workflows ("General CI options" above XXX).
```
Comment (guidezpl) [inline src/docs/deployment/cd.md:31]: i.e. you likely wouldn't use fastlane unless you were using Cirrus, GitHub Actions, Travis, or Gitlab

File: src/docs/deployment/android.md [hunk 3]
```diff
@@ -92,57 +92,66 @@ To find out the latest version, visit [Google Maven][].
...
-{{site.alert.end}}
+If you have an existing keystore, skip to the next step.
+If not, create one by either:
+* Following the [Android Studio key generation steps]({{site.android-dev}}/studio/publish/app-signing#sign-apk) 
+* Running the following at the command line:
+
+    On Mac/Linux, use the following command:
+
+    ```terminal
+    keytool -genkey -v -keystore ~/upload-keystore.jks -keyalg RSA -keysize 2048 -validity 10000 -alias upload
```
Comment (guidezpl) [inline src/docs/deployment/android.md:112]: standardize naming to upload-keystore.jks

File: src/docs/deployment/cd.md [hunk 4]
```diff
@@ -149,53 +163,41 @@ secrets in pull requests that you accept and merge.
...
-    * Shard your script to run on both Linux and macOS platforms.
-    * Remember to specify a dependency on Xcode for macOS (for example
-      `osx_image: xcode9.2`).
     * See [fastlane CI documentation][] for CI specific setup.
-    * During the setup phase, depending on the platform, make sure that:
-         * Bundler is available using `gem install bundler`.
-         * For Android, make sure the Android SDK is available and the `ANDROID_SDK_ROOT`
-           path is set.
+    * Shard your script to run on both Linux and macOS platforms.
+    * During the setup phase of the CI task:
```
Comment (sfshaza2) [inline src/docs/deployment/cd.md:168]:
```suggestion
    * During the setup phase of the CI task, do the following:
```

**Review by sfshaza2 — APPROVED** [review]
LGTM with a couple minor nits.

File: src/docs/deployment/cd.md [hunk 5]
```diff
@@ -7,11 +7,32 @@ Follow continuous delivery best practices with Flutter to make sure your
 application is delivered to your beta testers and validated on a frequent basis
 without resorting to manual workflows.
 
+
+## CI/CD Options
+
+There are a number of continuous integration (CI) and continuous delivery (CD)
+options available to help automate the delivery of your application.
+
+### All-in-one type of options, Flutter functionality built in XXX
```
Comment (sfshaza2) [inline src/docs/deployment/cd.md:16]:
```suggestion
### All-in-one options with built-in Flutter functionality
```

File: src/docs/deployment/cd.md [hunk 6]
```diff
@@ -7,11 +7,32 @@ Follow continuous delivery best practices with Flutter to make sure your
...
+### General CI options XXX
+* [Cirrus][]
+* [GitHub Actions][]
+    * Example: Flutter Gallery's [Github Actions workflows][]
+    * Example: [Github Action in Flutter Project][]
+* [Travis][]
+* [GitLab][]
+
+This guide shows how to set up fastlane and then integrate it with 
+your existing testing and continuous integration (CI) workflows ("General CI options" above XXX).
```
Comment (sfshaza2) [inline src/docs/deployment/cd.md:30]:
```suggestion
### Integrating fastlane with existing workflows

You can use fastlane with the following tooling:

* [GitHub Actions][]
    * Example: Flutter Gallery's [Github Actions workflows][]
    * Example: [Github Action in Flutter Project][]
* [Cirrus][]
* [Travis][]
* [GitLab][]

This guide shows how to set up fastlane and then integrate it with 
your existing testing and continuous integration (CI) workflows. 
For more information, see "Integrating fastlane with existing workflow".
```

**Review by sfshaza2 — APPROVED** [review]
Hopefully, the suggested headers make sense. I'm going to merge. Thanks for this PR! LGTM

**Comment by guidezpl** [comment]
Makes sense,thanks!

**Pull request merged**